sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosAsyncContainer.java

```diff
@@ -532,6 +534,102 @@ public String getId() {
         return withContext(context -> readItemInternal(itemId, requestOptions, itemType, context));
     }
 
+    /**
+     * Reads many documents.
+     *
+     * @param <T> the type parameter
+     * @param itemKeyList document id and partition key pair that needs to be read
+     * @param classType   class type
+     * @return a Mono with feed response of cosmos items
+     */
+    @Beta(Beta.SinceVersion.V4_4_0)
+    public <T> Mono<FeedResponse<T>> readMany(
+        List<Pair<String, PartitionKey>> itemKeyList,
+        Class<T> classType) {
+
+        return this.readMany(itemKeyList, null, classType);
+    }
+
+    /**
+     * Reads many documents.
+     *
+     * @param <T> the type parameter
+     * @param itemKeyList document id and partition key pair that needs to be read
+     * @param sessionToken the optional Session token - null if the read can be made without specific session token
+     * @param classType   class type
+     * @return a Mono with feed response of cosmos items
+     */
+    @Beta(Beta.SinceVersion.V4_4_0)
+    public <T> Mono<FeedResponse<T>> readMany(
+        List<Pair<String, PartitionKey>> itemKeyList,
+        String sessionToken,
```

Comment:
why do we need to expose session token here?
Did anyone ask for it? if not can we remove this?

Reply:
How would a customer who wants to use Session consistency do the ReadMany in this case?

Reply:
The idea was to start exposing this so that not other customers but Adobe take dependency on implementation package

Reply:
If this needs more discussions I will revert it here and leave it as is to not unblock the Walmart fix on this discusison.

Reply:
for the common session scenario session is resolved through internal sessionContainer. 
I would say passing a session token arg is an advanced use case.

Reply:
I would argue readMany is only necessary in advanced scenarios anyway. What is the harm in exposing sessionToken as parameter? Is the concern that you would prefer a special ReadManyRequestOptions with just SessionToken for now?

Reply:
Or don't want to decide on that model yet?

Reply:
I didn't want to decide on the contract/model. but this is beta anyway. 

so makes sense. let's keep it. resolving the comment.